napari/_vispy/vispy_camera.py

```diff
@@ -78,7 +78,10 @@ def center(self):
             center = self._view.camera.center[:2]
 
         # Switch from VisPy ordering to NumPy ordering
-        return tuple(center[::-1])
+        center = tuple(center[::-1])
+        if len(center) == 2:
+            center = (0.0,) + center
+        return center
```

Comment:
@sofroniewn can you clarify what's happening here? I find it rather disconcerting. Is center now always 3D and we have a dummy 0.0 when we only want 2D? Is this necessary behaviour? (ie could we instead use a Union type?) If so, I think the following code might be clearer, because there's only one place where we clip and append:

```python
if self._view.camera == self._3D_camera:
    center = tuple(self.view.camera.center)
else:
    # in 2D, we arbitrarily choose 0.0 as the center in z
    center = tuple(self._view.camera.center[:2]) + (0.0,)
# switch from VisPy xyz ordering to NumPy prc ordering
center = center[::-1]
return center
```

One question I have is why this value would ever become something different than 0, if this is a different camera object? ie why are we clipping with the `[:2]` and then adding (0.0,) vs keeping that value pinned at 0?

Reply:
Yeah, center is always length three, but we don't actually need this change here - even if we remove it when you do `camera.center = [20, 10]` say you'd get the value to be `[0, 20, 10]`. I prefer your code snippet though and will update